### PR TITLE
Improve docs regarding SPAs and improve the DEV watcher

### PR DESCRIPTION
**Pull request by clebert**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -453,6 +455,8 @@ exports.default = {
 };
 ```
 
+_Note: This feature can be configured in the same way for Lambda functions._
```
[the remark this answers is not in the document]
Comment (clebert) [reply README.md]: Done

File: src/express/start-dev-server.ts [hunk 2]
```diff
@@ -70,11 +70,7 @@ export async function startDevServer(init: DevServerInit): Promise<void> {
   app.listen(port, () => {
     logInfo(`Started DEV server: http://localhost:${port}`);
 
-    const localPaths = [...lambdaConfigs, ...s3Configs].map(
-      ({localPath}) => localPath
-    );
-
-    watch(localPaths).on('change', (changedLocalPath) => {
+    const watchListener = (changedLocalPath: string) => {
```
Comment (unstubbable) [inline src/express/start-dev-server.ts:73]:
```suggestion
    const handleLocalPathChanges = (changedLocalPath: string) => {
```

File: src/express/start-dev-server.ts [hunk 3]
```diff
@@ -50,7 +50,7 @@ export async function startDevServer(init: DevServerInit): Promise<void> {
 
   for (const routeConfig of sortRouteConfigs([
     ...lambdaConfigs,
-    ...resolveS3FileConfigs(s3Configs),
+    ...resolveS3FileConfigs(s3Configs, true),
```
Comment (unstubbable) [inline src/express/start-dev-server.ts:53]: options argument

Comment (unstubbable) [reply src/express/start-dev-server.ts:53]: `devMode`?

File: README.md [hunk 4]
```diff
@@ -423,12 +423,14 @@ associated with the stack using the `aws-simple upload [options]` CLI command.
 The optionally specified `bucketPath` or, if not specified, the `publicPath` is
 used as the S3 object key._
 
-#### Configure A Single-page Application
+#### Configure A Single-Page Application
```
Comment (unstubbable) [inline README.md:426]: add "SPA" in brackets

Comment (unstubbable) [reply README.md:426]: move out of s3 section into its own section

File: src/utils/resolve-s3-file-configs.ts [hunk 5]
```diff
@@ -21,18 +24,24 @@ export function resolveS3FileConfigs(s3Configs: S3Config[]): S3FileConfig[] {
         );
       }
 
-      const filenames = readdirSync(localPath)
-        .map((filename) => path.resolve(path.join(localPath, filename)))
-        .filter((filename) => lstatSync(filename).isFile());
+      try {
+        const filenames = readdirSync(localPath)
```
Comment (unstubbable) [inline src/utils/resolve-s3-file-configs.ts:28]: check for existence instead of try/catch

**Pull request merged**